src/App.tsx

```diff
@@ -15,35 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Params = {
+  sortField: SortType;
+  isReversed: boolean;
+};
+
+enum SortType {
+  default = '',
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+function getPreparedGoods(goods: string[], params: Params): string[] {
+  const { sortField, isReversed } = params;
+
+  const preparedGoods = [...goods].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.alphabet:
+        return good1.localeCompare(good2);
+      case SortType.length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(isReversed || sortField) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.default);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
```

Comment:
There is an unnecessary nested `<ul>` tag here. It would be better to remove the inner `<ul>` to maintain a proper HTML structure. The `visibleGoods.map` should directly render `<li>` elements within a single `<ul>`.